bindings/rust/s2n-tls/src/pool.rs

```diff
@@ -117,6 +117,19 @@ impl<T: Pool> Pool for Arc<T> {
     }
 }
 
+/// A pool of Connections. Not a pool of Configs.
+///
+/// Instead of allocating and freeing new `Connection`s, this struct allows them
+/// to be "wiped" and reused.
+///
```

Comment:
This entire module is about pooling connections. The Pool interface is specifically for polling connections, not for pooling random objects. This documentation probably either belongs at the module level, or on Pool. I'd expect the ConfigPool-specific documentation to then just describe how it implements connection reuse (by creating connections with the same config).

Reply:
My concern is that
1. few customers actually implement their own pool, and instead everyone just uses `ConfigPool`
2. therefore any of the documentation on `Pool` is must less visible than the documentation on `ConfigPool`.

Reply:
I don't think we should put module-level documentation on a struct just because it's the most popular feature of the module. We could link to the module-level documentation from the struct though if you're worried about discovery-- I've read a lot of docs that do that.